### PR TITLE
Add `uploadedAt` to tokens

### PR DESCRIPTION
**Pull request by rogierslag**
Using an `uploadedAt` allows us to keep multiple services instances in sync easier (we can do selective pulls from AWS). However adding that required quite a bunch of scaffolding: for all existing installations this information is missing, but it is required at a later stage.

As such the application now ships with a `appchangelog` where javascript based migration can be executed. The app changes are performed on a single instance, and after the database migrations.

This ensures that existing installations are back-filled with the required data for later

### CODE REVIEW

File: src/migrations/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+// eslint-disable-next-line no-process-env
+const APP_INSTANCE = process.env.NODE_APP_INSTANCE;
+
+function canNodeRunMigration() {
+  // Run when not in PM2 or on the first instance only
+  return !APP_INSTANCE || APP_INSTANCE === '0';
+}
+
+function startMigration(method) {
```
Comment (AlexKolpa) [inline src/migrations/index.js]: `method` => `migration`?

File: src/migrations/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  return new Promise(resolve => {
+    if (!canNodeRunMigration()) {
+      resolve();
+      return;
+    }
+    const reschedule = () => setTimeout(async () => {
```
Comment (AlexKolpa) [inline src/migrations/index.js]: What's the reason for running this in a timeout? It seems a bit weird to block 2.5s on boot for that.

File: src/migrations/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      const done = await method();
```
Comment (AlexKolpa) [inline src/migrations/index.js]:
```javascript
try {
 await method();
} catch(e) {
  reject(e);
}
```
Right now I'm pretty sure the error is black-holed.

File: src/migrations/index.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      if (!done) {
+        reschedule();
+      } else {
+        resolve();
+      }
+    }, 2500);
```
Comment (AlexKolpa) [inline src/migrations/index.js]: Imo it's not the responsibility of this logic to assume a migration function is only a partial migration. The migration function should resolve iff it's actually complete, and handle any batching (which this seems to do) internally. I would expect any failure to result in a rejected promise.

File: src/migrations/index.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    case 'uploadedAt':
+      try {
+        await startMigration(populateUploadedAt);
+        await database.markAppMigrationAsCompleted(next);
+        log('info', `Finished migration for ${next}`);
+      } catch (e) {
+        log('warn', `Migration for ${next} threw in the process. This migration will be retried.`);
+      }
+      break;
+    case null:
```
Comment (AlexKolpa) [inline src/migrations/index.js]: Maybe throw an error for `default` to prevent migration paths being missed?

Comment (rogierslag) [reply src/migrations/index.js]: fair enough and I just realized I already handle the null above. Plus this code is getting quite complicated actually for now reason

**Pull request merged**